src/prop/minisat/simp/SimpSolver.cc

```diff
@@ -51,23 +51,22 @@ SimpSolver::SimpSolver(Env& env,
                        prop::TheoryProxy* proxy,
                        context::Context* context,
                        context::UserContext* userContext,
-                       prop::PropPfManager* ppm,
                        bool enableIncremental)
-    : Solver(env, proxy, context, userContext, ppm, enableIncremental),
+    : Solver(env, proxy, context, userContext, enableIncremental),
       grow(opt_grow),
       clause_lim(opt_clause_lim),
       subsumption_lim(opt_subsumption_lim),
       simp_garbage_frac(opt_simp_garbage_frac),
       use_asymm(opt_use_asymm),
       // make sure this is not enabled if unsat cores or proofs are on
-      use_rcheck(opt_use_rcheck && !options().smt.produceUnsatCores && !ppm),
+      use_rcheck(opt_use_rcheck && !options().smt.produceUnsatCores),
       merges(0),
       asymm_lits(0),
       eliminated_vars(0),
       elimorder(1),
       use_simplification(
           options().prop.minisatSimpMode != options::MinisatSimpMode::NONE
-          && !enableIncremental && !options().smt.produceUnsatCores && !ppm),
+          && !enableIncremental && !options().smt.produceUnsatCores),
```

Comment:
I'm not sure this can be done here. `use_simplification` is used in the body below. If proofs are enabled, we would still execute the below code on line 85. We should be able to use `env.isSatProofProducing()` here instead. @ajreynol does that sound right?

Reply:
Yes, checking `env.isSatProofProducing` is preferred to checking `!ppm`.